src/main/java/uk/gov/hmcts/reform/migration/service/migrate/MultipleHearingsOutcomeMigration.java

```diff
@@ -0,0 +1,51 @@
+package uk.gov.hmcts.reform.migration.service.migrate;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.domain.hmc.CaseHearing;
+import uk.gov.hmcts.reform.migration.hmc.HmcHearingsApiService;
+import uk.gov.hmcts.reform.migration.repository.CaseLoader;
+import uk.gov.hmcts.reform.migration.service.HearingOutcomeService;
+
+import java.util.List;
+import java.util.Map;
+
+@Service
+@Slf4j
+@ConditionalOnProperty(value = "migration.multipleHearingsOutcomes.enabled", havingValue = "true")
+public class MultipleHearingsOutcomeMigration extends CaseOutcomeMigration {
+
+    private final HmcHearingsApiService hmcHearingsApiService;
+
+    private final Map<String, String> caseMap = new CaseLoader(encodedDataString).findCasesWithHearingID();
```

Comment:
I was thinking instead of populating the map here and decoding the string twice, we override the fetchCasesToMigrate method so that it calls findCasesWithHearingID
that method can be updated so that it returns the list of cases but also populates the map